graphql_compiler/compiler/ir_lowering_common.py

```diff
@@ -1,24 +1,26 @@
 # Copyright 2017 Kensho Technologies, Inc.
 """Language-independent IR lowering and optimization functions."""
 
-from funcy.py2 import pairwise
-
-from .blocks import ConstructResult, Filter
+from .blocks import ConstructResult, Filter, Fold, Unfold
 from .expressions import (BinaryComposition, ContextField, ContextFieldExistence, FalseLiteral,
                           NullLiteral, TrueLiteral)
 from .helpers import validate_safe_string
 
 
 def merge_consecutive_filter_clauses(ir_blocks):
     """Merge consecutive Filter(x), Filter(y) blocks into Filter(x && y) block."""
+    if not ir_blocks:
+        return ir_blocks
```

Comment:
This seems interesting. This _won't_ just explicitly return `None`, but rather `[]` _or_ `None` (depending on whether a non-existent `ir_blocks` array is even possible).

Reply:
Right -- but `ir_blocks` can't ever be `None`. I can add an assertion for this, though it seems unnecessary since:
- `ir_blocks` is generated by the compiler front-end which is super well-tested;
- if it were `None`, this would fail in an obvious way in 10 other places after this.